app/models/katello/host/subscription_facet.rb

```diff
@@ -80,7 +84,7 @@ def consumer_attributes
             product
           end
         end
-        attrs
+        HashWithIndifferentAccess.new(attrs)
```

Comment:
Switching to HashWithIndifferentAccess will eliminate the need for those .try() in other methods. The data coming in via PUT /hosts were symbols not strings so the check 'guestIds' was failing to trigger the hypervisor=true check.